rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -348,9 +349,52 @@ public boolean nodeIsAvailable(@HeaderParam("sessionid") String sessionId, @Quer
         return rm.getExistingNodeSourcesList();
     }
 
+    @Override
+    @POST
+    @Path("nodesource")
+    @Produces("application/json")
+    public NSState defineNodeSource(@HeaderParam("sessionid") String sessionId,
+            @FormParam("nodeSourceName") String nodeSourceName,
+            @FormParam("infrastructureType") String infrastructureType,
+            @FormParam("infrastructureParameters") String[] infrastructureParameters,
+            @FormParam("infrastructureFileParameters") String[] infrastructureFileParameters,
+            @FormParam("policyType") String policyType, @FormParam("policyParameters") String[] policyParameters,
+            @FormParam("policyFileParameters") String[] policyFileParameters,
+            @FormParam("nodesRecoverable") String nodesRecoverable) throws NotConnectedException {
+        ResourceManager rm = checkAccess(sessionId);
+        Object[] infraParams = new Object[infrastructureParameters.length + infrastructureFileParameters.length];
+        Object[] policyParams = new Object[policyParameters.length + policyFileParameters.length];
+        NSState nsState = new NSState();
+
+        prepareInfraParams(infrastructureType, infrastructureParameters, infrastructureFileParameters, rm, infraParams);
+
+        preparePolicyParams(policyType, policyParameters, policyFileParameters, rm, policyParams);
+
+        try {
+            nsState.setResult(rm.defineNodeSource(nodeSourceName,
+                                                  infrastructureType,
+                                                  infraParams,
+                                                  policyType,
+                                                  policyParams,
+                                                  Boolean.parseBoolean(nodesRecoverable))
+                                .getBooleanValue());
+        } catch (RuntimeException ex) {
+            nsState.setResult(false);
+            nsState.setErrorMessage(cleanDisplayedErrorMessage(ex.getMessage()));
+            nsState.setStackTrace(StringEscapeUtils.escapeJson(getStackTrace(ex)));
+
+        } finally {
+            return nsState;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this return statement from this finally block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1143)

Reply:
:+1: 

Can you remove the return from the finally block. The next developer might overlook that return, while adding a new one somewhere else.

Reply:
yes, it can be outside of the try/catch

Reply:
done